datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -824,6 +852,30 @@ mod tests {
             Operator::Lt,
             DataType::Date64
         );
+        test_coercion_binary_rule!(
+            DataType::Utf8,
+            DataType::Time32(TimeUnit::Second),
+            Operator::Eq,
+            DataType::Time32(TimeUnit::Second)
+        );
+        test_coercion_binary_rule!(
+            DataType::Utf8,
+            DataType::Time32(TimeUnit::Millisecond),
+            Operator::Eq,
+            DataType::Time32(TimeUnit::Millisecond)
+        );
+        test_coercion_binary_rule!(
+            DataType::Utf8,
+            DataType::Time64(TimeUnit::Microsecond),
+            Operator::Eq,
+            DataType::Time64(TimeUnit::Microsecond)
+        );
+        test_coercion_binary_rule!(
```

Comment:
**suggestion:** Should also probably check coercion which are not equal

Reply:
Not sure if I understood here. Can you please give me an example? 

Is it like type coercion between `Time64(Microsecond)` and `Time64(Nanosecond)`? If that is the case, in fact I have not implemented it, as I wanted to keep the scope of the PR really simple. But I am not sure it is this what you meant...